src/control/cmd/dmg/pretty/storage_nvme.go

```diff
@@ -165,32 +165,38 @@ func parseNvmeFormatResults(inResults storage.NvmeControllers) storage.NvmeContr
 	parsedResults := make(storage.NvmeControllers, 0, len(inResults))
 	for _, result := range inResults {
 		if result.PciAddr != storage.NilBdevAddress {
-			// ignore skip results
 			parsedResults = append(parsedResults, result)
 		}
 	}
 
 	return parsedResults
 }
 
-func printNvmeFormatResults(devices storage.NvmeControllers, out io.Writer, opts ...PrintConfigOption) error {
-	if len(devices) == 0 {
+func printNvmeFormatResults(ctrlrs storage.NvmeControllers, out io.Writer, opts ...PrintConfigOption) error {
+	if len(ctrlrs) == 0 {
 		fmt.Fprintln(out, "\tNo NVMe devices found")
 		return nil
 	}
 
 	pciTitle := "NVMe PCI"
 	resultTitle := "Format Result"
+	rolesTitle := "Role(s)"
 
-	formatter := txtfmt.NewTableFormatter(pciTitle, resultTitle)
+	formatter := txtfmt.NewTableFormatter(pciTitle, resultTitle, rolesTitle)
 	formatter.InitWriter(out)
 	var table []txtfmt.TableRow
 
-	sort.Slice(devices, func(i, j int) bool { return devices[i].PciAddr < devices[j].PciAddr })
+	sort.Slice(ctrlrs, func(i, j int) bool { return ctrlrs[i].PciAddr < ctrlrs[j].PciAddr })
 
-	for _, device := range parseNvmeFormatResults(devices) {
-		row := txtfmt.TableRow{pciTitle: device.PciAddr}
-		row[resultTitle] = device.Info
+	for _, ctrlr := range parseNvmeFormatResults(ctrlrs) {
+		row := txtfmt.TableRow{pciTitle: ctrlr.PciAddr}
+		row[resultTitle] = ctrlr.Info
+		roles := "NA"
+		// Assumes that all SMD devices on a controller have the same roles.
```

Comment:
Is this really a safe assumption?

Reply:
currently yes, we don't have capability to have separate blobstores on any single SSD, so all devices will have the same role combinations

Reply:
If this assumption is likely to hold true going forward, I don't have a problem with it. Just looking to avoid churn when possible.